docs/manual/contributing.rst

```diff
@@ -25,14 +25,21 @@ has two qualities:
 
 Ensure to include any appropriate log entries from
 ``~/.local/share/qtile/qtile.log`` and/or ``~/.xsession-errors``!
+Sometimes, an ``xtrace`` is requested. If that is the case, refer to
+:ref:`capturing an xtrace <capturing-an-xtrace>`.
+
 
 Writing code
 ============
 
 To get started writing code for Qtile, check out our guide to :ref:`hacking`.
-
 A more detailed page on creating widgets is available :ref:`here <widget-creation>`.
 
+.. important::
+
+    Use a separate **git branch** to make rebasing easy. Ideally, you would
+    ``git checkout -b <my_feature_branch_name>`` before starting your work.
```

Comment:
I wonder if we shouldn't just link to a guide about git branching here, talking about git rebase -i, that git push -f is ok, etc?

Reply:
I had a look around and although there are lots of guides for various combinations of git commands, it might be more straightforward (and less confusing to new users) to just mention a few commands to start them on searching around themselves. We can add a sentence saying when `git rebase -i` might be useful and that `git push -f` can be used to update the commits on their PR. From there they might be good to go.